src/datepicker/adapters/ngb-date-native-adapter.ts

```diff
@@ -9,6 +10,16 @@ export class NgbDateNativeAdapter extends NgbDateAdapter<Date> {
   }
 
   toModel(date: NgbDateStruct): Date {
-    return date && date.year && date.month ? new Date(date.year, date.month - 1, date.day, 12) : null;
+    if (date && isNumber(date.year) && isNumber(date.month) && isNumber(date.day)) {
+      const jsDate = new Date(date.year, date.month - 1, date.day, 12);
+
+      // this is done avoid 30 -> 1930 conversion
+      if (!isNaN(jsDate.getTime())) {
```

Comment:
Couldn't we test for `date.year` < 100 (or whatever cut off point for conversion is)?

Reply:
Actually I was doing it to check that the `jsDate` is a valid date object. I guess more correct version would be:

```ts
if (jsDate instanceof Date && !isNan(jsDate) && date.year >= 0 && date.year <= 99) { ... }
```

But I kinda can omit `instanceof` part.

Do you want it to be like this ?
```ts
if (!isNan(jsDate) && date.year >= 0 && date.year <= 99) { ... }
```